worlds/saving_princess/Client.py

```diff
@@ -0,0 +1,215 @@
+        with open("original_data.win", "rb") as data:
+            patched_data = bsdiff4.patch(data.read(), download.read())
+        with open("data.win", "wb") as data:
+            data.write(patched_data)
+        logging.info("Done!")
+
+
+def is_install_valid() -> bool:
+    """Checks that the mandatory files that we cannot replace do exist in the current folder"""
+    for file_name, expected_hash in file_hashes.items():
+        if not os.path.exists(file_name):
+            return False
+        with open(file_name, "rb") as clean:
+            current_hash = hashlib.md5(clean.read()).hexdigest()
+        if not secrets.compare_digest(current_hash, expected_hash):
+            return False
+    return True
+
+
+def install() -> None:
+    """Extracts all the game files into the mod installation folder"""
+    logging.info("Mod installation missing or corrupted, proceeding to reinstall.")
+    # get the cab file and extract it into the installation folder
+    with open(SavingPrincessWorld.settings.exe_path, "rb") as exe:
+        # find the cab header
+        logging.info("Looking for cab archive inside exe.")
+        cab_found: bool = False
+        while not cab_found:
+            cab_found = exe.read(1) == b'M' and exe.read(1) == b'S' and exe.read(1) == b'C' and exe.read(1) == b'F'
+        exe.read(4)  # skip reserved1, always 0
+        cab_size: int = int.from_bytes(exe.read(4), "little")  # read size in bytes
+        exe.seek(-12, 1)  # move the cursor back to the start of the cab file
+        logging.info(f"Archive found at offset {hex(exe.seek(0, 1))}, size: {hex(cab_size)}.")
+        logging.info("Extracting cab archive from exe.")
+        with open("saving_princess.cab", "wb") as cab:
+            cab.write(exe.read(cab_size))
+    logging.info("Extracting files from cab archive.")
+    if Utils.is_windows:
+        subprocess.run(["Extrac32", "/Y", "/E", "saving_princess.cab"])
+    else:
+        if shutil.which("wine") is not None:
+            subprocess.run(["wine", "Extrac32", "/Y", "/E", "saving_princess.cab"])
+        elif shutil.which("7z") is not None:
+            subprocess.run(["7z", "e", "saving_princess.cab"])
+        else:
+            error = "Could not find neither wine nor 7z.\n\nPlease install either the wine or the p7zip package."
+            messagebox.showerror("Missing package!", f"Error: {error}")
+            raise RuntimeError(error)
+    os.remove("saving_princess.cab")  # delete the cab file
+    shutil.copyfile("data.win", "original_data.win")  # and make a copy of data.win
+    if os.path.exists("versions.json"):
+        os.remove("versions.json")  # to force reinstall of the assets
+    logging.info("Done!")
+
+
+def launch() -> None:
+    """Check the mod installation, then launch it"""
+    os.chdir(SavingPrincessWorld.settings.install_folder)
+
+    # check that the mod installation is valid
+    valid_install = is_install_valid()
+    if not valid_install:
```

Comment:
```suggestion
    if not is_install_valid():
```
No need to hold onto this value since it's not used elsewhere